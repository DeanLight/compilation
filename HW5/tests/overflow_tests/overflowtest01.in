void main(){
	byte byte1;
	byte byte2;


	printi(byte1 - 1);
	print("\n");

	byte2=byte1
	printi(byte2);
	print("\n");

	printi(16*16);
	print("\n");

	printi(byte2+1);
	print("\n");

	printi(byte2+3);
	print("\n");

	printi(255 b + 255 b);
	print("\n");

	int maxint=2147483647
	int minint=0-2147483648
	int another_minint=2147483648
	int another_maxint=0-2147483649
	int num1;
	int num2;



	printi(255 + 255 );
	print("\n");

	printi(maxint);
	print("\n");

	printi(another_maxint );
	print("\n");

	printi(minint );
	print("\n");

	printi(another_minint );
	print("\n");

	num1=maxint+1

	printi(num1);
	print("\n");

	num2=maxint+maxint+2

	printi(num2);
	print("\n");

	num2=maxint+1+num1
	printi(num2);
	print("\n");

	printi(100000*100000);
	print("\n");

	byte1=200;
	byte2=100;

	int1=byte1+byte2;
	printi(int1);
	print("\n");

	int1=byte1+byte2;
	printi(int1);
	print("\n");

}
