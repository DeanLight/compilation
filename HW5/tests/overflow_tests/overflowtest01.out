Loaded: ./exceptions.s
255
255
0
0
2
254
510
2147483647
2147483647
-2147483648
-2147483648
-2147483648
1
1
1410065408
45
300